core/worker.go

```diff
@@ -976,19 +976,32 @@ func (w *worker) commit(env *environment, interval func(), update bool, start ti
 	return nil
 }
 
-// AddPendingBlockBody adds an entry in the lru cache for the given block root
+// GetPendingBlockBodyKey takes a header and hashes all the Roots together
+// and returns the key to be used for the pendingBlockBodyCache.
+func (w *worker) GetPendingBlockBodyKey(header *types.Header) (hash common.Hash) {
+	nodeCtx := common.NodeLocation.Context()
+	hdata := types.PendingBodyKey{
+		UncleHash: header.UncleHash(nodeCtx),
+		TxHash:    header.TxHash(nodeCtx),
+		EtxHash:   header.EtxHash(nodeCtx),
+	}
+	return hdata.Hash()
+}
```

Comment:
In addition to the comment above, you also do not need to (and should not) manually index into the `UncleHash()`, `TxHash()`, and `EtxHash()` methods. This function should really just be:
```
func (w *worker) GetPendingBlockBodyKey(header *types.Header) (hash common.Hash) {
	return rlpHash([]interface{
		UncleHash: header.UncleHash(),
		TxHash:    header.TxHash(),
		EtxHash:   header.EtxHash(),
	})
}
```

Reply:
And it should be a private internal method, since the only place this should ever be called is for the associated Read/Write/Delete methods

Reply:
made it a private method and RLP is done in the function itself